### ISSUE TITLE
Magento installation process disregards cache id_prefix settings defined in env.php (without specify settings via installer options)

### ISSUE THREAD

**Issue by cmacdonald-au** [magento/magento2#22253]
### Preconditions (*)

1. A copy of Magento, (either installed, or not) >= 2.3.1
2. An `app/etc/env.php` file that contains cache settings, (example attached, specifically an `id_prefix` not based on path must be set).
3. Execute `./bin/magento setup:install` or `./bin/magento setup:install --cleanup-database` (with no other parameters specified).

This behaviour appears to be a side effect of the changes introduced in #18641

### Steps to reproduce (*)

Either:
1.  starting from a fresh (uninstalled) Magento2 installation; Take an `app/etc/env.php` from a previously installed system and: 
 - remove `install_date`.
 - change the cache `id_prefix` values to something not based on path.
2.  using a previously installed version of magento, change the cache `id_prefix` values to something not based on path and flush cache.

Then:
1. Execute `./bin/magento setup:install` (use `--cleanup-database` to test against a previously installed copy)
2. compare the updated `app/etc/env.php` with the original and you will see the cache related `id_prefix` values have changed.

### Expected result (*)

1. The `id_prefix` that was provided within `env.php` prior to installation must be unchanged

### Actual result (*)

1. The `id_prefix` has been replaced by the output of `Magento\Setup\Model\ConfigOptionsList\Cache::generateCachePrefix()`

### Supporting information
Example "starter" `env.php` file: [preinstall-env.php.txt](https://github.com/magento/magento2/files/3061725/preinstall-env.php.txt)

Resulting diff;
```
➜  m2 git:(2.3-develop) diff -ub app/etc/preinstall-env.php.txt app/etc/env.php
--- app/etc/preinstall-env.php.txt	2019-04-10 12:15:51.000000000 +1000
+++ app/etc/env.php	2019-04-10 12:28:39.000000000 +1000
@@ -83,7 +83,7 @@
     'cache' => [
         'frontend' => [
             'default' => [
-                'id_prefix' => 'm2l_',
+                'id_prefix' => '74c_',
                 'backend' => 'Cm_Cache_Backend_Redis',
                 'backend_options' => [
                     'server' => '127.0.0.1',
@@ -95,7 +95,7 @@
                 ]
             ],
             'page_cache' => [
-                'id_prefix' => 'm2l_',
+                'id_prefix' => '74c_',
                 'backend' => 'Cm_Cache_Backend_Redis',
                 'backend_options' => [
                     'server' => '127.0.0.1',
@@ -127,6 +127,6 @@
         'vertex' => 1
     ],
     'install' => [
-        'date' => ''
+        'date' => 'Wed, 10 Apr 2019 02:28:34 +0000'
     ]
 ];
```

### Why this matters..
Our CI/CD processes manage the propagation of settings upwards and outwards. This extends to unattended installations as well as new production deployments.

All of our systems have the opportunity to share cache instances (as either cost or convenience measures) and - as per the database table prefix setting - we rely on being able to control the cache `id_prefix` settings to ensure they support the purpose they were introduced for.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99209` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @cmacdonald-au. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24377` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
